### ISSUE TITLE
`(mapcat identity)` cannot be used as a `xf` with Promesa channels 

### ISSUE THREAD

**Issue by fr33m0nk** [funcool/promesa#129]
Hi,

I was toying around with latest release of Promesa and found a possible bug.
`(mapcat identity)` cannot be used as a transformer with Promesa channels.

Below is the behavior with Promesa channels:
```clojure
(def promesa-chan (sp/chan 100 (mapcat identity)))
;;=> #'clj-actor.promesea.core/promesa-chan

(sp/put! promesa-chan [1 2 3 4 5])

Exception in thread "nREPL-session-a5c4ff13-7be0-4f1a-bcf8-0d8ffdeee2ad" java.lang.IllegalArgumentException: No implementation of method: :-offer! of protocol: #'promesa.protocols/IBuffer found for class: java.lang.Boolean
	at clojure.core$_cache_protocol_fn.invokeStatic(core_deftype.clj:584)
	at clojure.core$_cache_protocol_fn.invoke(core_deftype.clj:576)
	at promesa.protocols$eval11897$fn__11909$G__11886__11916.invoke(protocols.cljc:87)
	at promesa.exec.csp.channel$add_fn.invokeStatic(channel.cljc:329)
	at promesa.exec.csp.channel$add_fn.invoke(channel.cljc:325)
	at clojure.core$preserving_reduced$fn__8848.invoke(core.clj:7682)
	at clojure.lang.PersistentVector.reduce(PersistentVector.java:343)
	at clojure.core$reduce.invokeStatic(core.clj:6885)
	at clojure.core$cat$fn__8851.invoke(core.clj:7693)
	at clojure.core$map$fn__5931$fn__5932.invoke(core.clj:2759)
	at promesa.exec.csp.channel$chan$add_fn__12827.invoke(channel.cljc:372)
	at promesa.exec.csp.channel.Channel._put_BANG_(channel.cljc:241)
	at promesa.exec.csp$put.invokeStatic(csp.clj:92)
	at promesa.exec.csp$put.invoke(csp.clj:81)
	at promesa.exec.csp$put_BANG_.invokeStatic(csp.clj:108)
	at promesa.exec.csp$put_BANG_.invoke(csp.clj:105)
	at clj_actor.promesea.core$eval13713.invokeStatic(form-init3957983138384948556.clj:1)
	at clj_actor.promesea.core$eval13713.invoke(form-init3957983138384948556.clj:1)
	at clojure.lang.Compiler.eval(Compiler.java:7194)
	at clojure.lang.Compiler.eval(Compiler.java:7149)
	at clojure.core$eval.invokeStatic(core.clj:3215)
	at clojure.core$eval.invoke(core.clj:3211)
	at nrepl.middleware.interruptible_eval$evaluate$fn__968$fn__969.invoke(interruptible_eval.clj:87)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$with_bindings_STAR_.invokeStatic(core.clj:1990)
	at clojure.core$with_bindings_STAR_.doInvoke(core.clj:1990)
	at clojure.lang.RestFn.invoke(RestFn.java:425)
	at nrepl.middleware.interruptible_eval$evaluate$fn__968.invoke(interruptible_eval.clj:87)
	at clojure.main$repl$read_eval_print__9206$fn__9209.invoke(main.clj:437)
	at clojure.main$repl$read_eval_print__9206.invoke(main.clj:437)
	at clojure.main$repl$fn__9215.invoke(main.clj:458)
	at clojure.main$repl.invokeStatic(main.clj:458)
	at clojure.main$repl.doInvoke(main.clj:368)
	at clojure.lang.RestFn.invoke(RestFn.java:1523)
	at nrepl.middleware.interruptible_eval$evaluate.invokeStatic(interruptible_eval.clj:84)
	at nrepl.middleware.interruptible_eval$evaluate.invoke(interruptible_eval.clj:56)
	at nrepl.middleware.interruptible_eval$interruptible_eval$fn__999$fn__1003.invoke(interruptible_eval.clj:152)
	at clojure.lang.AFn.run(AFn.java:22)
	at nrepl.middleware.session$session_exec$main_loop__1067$fn__1071.invoke(session.clj:202)
	at nrepl.middleware.session$session_exec$main_loop__1067.invoke(session.clj:201)
	at clojure.lang.AFn.run(AFn.java:22)
	at java.base/java.lang.Thread.run(Thread.java:1591)

;;=> true
```

Below is behavior with `core.async` channel
```clojure
(def async-chan (a/chan 100 (mapcat identity)))
;;=> #'clj-actor.promesea.core/async-chan
(a/>!! async-chan [1 2 3 4 5])
;;=> true
(a/<!! async-chan)
;;=> 1
```

**Comment by niwinz** [comment]
yep, seems like a bug, will look on it :D

**Comment by niwinz** [comment]
This should be fixed in 10.0.575, thanks for report
